### PR TITLE
Translate /i regexes to RE2 regex syntax

### LINKED ISSUE #119: Incorrect handling of regex options (/i, /m) for where queries
The translation to RethinkDB's RE2 syntax isn't correct. I will send a pull request shortly.

### PR DESCRIPTION
**Pull request by mbrock**
Fixes #119.

### CODE REVIEW

File: lib/no_brainer/criteria/where.rb [hunk 1]
```diff
@@ -285,11 +285,22 @@ def parse_multi_value(op, value, options={})
...
     when Range  then instantiate_binary_op(key, :between, value)
-    when Regexp then instantiate_binary_op(key, :match, value.inspect[1..-2])
+    when Regexp then instantiate_binary_op(key, :match, translate_regexp_to_re2_syntax(value))
     else instantiate_binary_op(key, :eq, value)
     end
   end
 
+  def translate_regexp_to_re2_syntax(value)
+    options = ""
+    options << "i" if 0 != value.options & Regexp::IGNORECASE
```
Comment (nviennot) [inline lib/no_brainer/criteria/where.rb]: `0 != x` -> `x != 0` (yoda comparison otherwise)

**Comment by mbrock** [comment]
Excellent question. I couldn't actually figure out what's going on with that flag.

From the RE2 syntax reference:

```
m	multi-line mode: ^ and $ match begin/end line in addition to begin/end text (default false)
s	let . match \n (default false)
```

But in my testing, it seemed as if these options were defaulting to true, though I couldn't see anything in the RethinkDB source code to indicate this... I got confused. Let me play around a bit more.

BTW, note that currently, the regex `foo/i` compiles into the string "foo/", which is broken.

File: lib/no_brainer/criteria/where.rb [hunk 2]
```diff
@@ -285,11 +285,22 @@ def parse_multi_value(op, value, options={})
...
+
+    if options.empty?
+      value.source
+    else
+      "(?#{options}:#{value.source})"
```
Comment (nviennot) [inline lib/no_brainer/criteria/where.rb]: From https://code.google.com/p/re2/wiki/Syntax it seems that we should rename `options` to `flags.

From the example on http://www.rethinkdb.com/api/ruby/match/ it seems that the parenthesis go around the flags, not the whole thing.

From the re2 docs:

> `(?flags)`	set flags within current group; non-capturing
> `(?flags:re)`	set flags during re; non-capturing

I don't understand the difference. Do you?

**Comment by mbrock** [comment]
Good points. Regarding the flags syntax, I don't think there is a difference in this case, but I could imagine that using the former might possibly give better error messages in some cases, or something.

**Comment by mbrock** [comment]
I've assumed that we want to ensure that `Regexp#match` matches the behavior of the RethinkDB query. This means always adding the `m` flag.

**Comment by nviennot** [comment]
your PR looks good to me, and I confirm your findings.

also, I wanted to add a "match" keyword to be explicit so that we can pass a raw RE2 string, but I can't define a match() method on symbol it seems :(

**Comment by nviennot** [comment]
Thank you :) :) :)

**Comment by mbrock** [comment]
Cool! By the way, I discovered this because rails_admin generates case-insensitive regexes for searching. It's quite fun to see rails_admin working with nobrainer. I'm finishing up the adapter for searching and filtering now. :smile:

**Comment by nviennot** [comment]
Well done!! I don't see your rails_admin fork though, where is it?

BTW, The RethinkDB guys will probably be happy to showcase it

**Pull request closed without merging**